### PR TITLE
Add Spotify Playlist Integration and Song Management to add song to Spotify Playlist in Trending Page

### PR DESCRIPTION
**Pull request by bumigumi**
TLDR: 
Adds dropdown next to "like" button on Trending page and dropdown populates with user's Spotify Playlists. Upon choosing a playlist and clicking "like", the song gets added to the chosen playlist.

Backend:
Implemented get-user-playlists endpoint to fetch user playlists from Spotify, implemented search-track endpoint to get song id based on song's name and artist, implemented add-to-playlist endpoint to add a song to a chosen playlist in spotify-routes.js.

Frontend:
Updated renderTrending to fetch and display playlists dynamically and added event listeners for dropdown selections.
Also integrated API calls for track search, playlist population to dropdown, and playlist addition in trending-script.js and made sure the dropdown looks good in css.

### CODE REVIEW

**Comment by cynthiaraj501** [comment]
Great feature addition! Looks all set to me.
